### ISSUE TITLE
Doesn't seem to work on GTK4:4.16

### ISSUE THREAD

**Issue by Roonil** [wmww/gtk4-layer-shell#50]
Using Endeavour OS with latest packages on the system (Laptop, CPU: Intel i7-9750H, external GPU: NVIDIA GTX 1650 Ti Mobile). Upon fresh installation, if latest GTK4 version is used, none of the example apps seem to work. It throws an error by Gdk-Message that says "Error 22 (Invalid argument) dispatching to Wayland display". However, if I roll back to a previous version such as GTK4:4.14.4-1, the example apps run successfully. I've also tried recompiling the examples at these version changes but I get the same result. This also happens on one of my friend's laptop (Endeavour OS, latest packages, AMD CPU, with the same external GPU as mine). The same error is thrown even when the internal or external GPU is used for rendering (using the DRI_PRIME, __NV_OFFLOAD) commands.

**Comment by wmww** [comment]
Oh no. I am unable to reproduce on Arch with `gtk4 1:4.16.3-1` and `gtk4-layer-shell 1.0.3-2` on `sway 1:1.9-5`.

What Wayland compositor/desktop environment are you using, and what version? That sort of error is generally caused by a mismatch in logic between the app/toolkit and compositor.

If anyone else sees this on a different distro please reply with what versions of things you're using, including wayland compositor. The more data I have on where this is appearing the easier it is to reproduce.

**Comment by wmww** [comment]
I've fixed a couple issues just now. A fresh build from the latest main might be fixed.

**Comment by Roonil** [comment]
I tested it on KDE Plasma 6.2.1. I also installed Hyprland (v0.44.1) just now, and if I use `GSK_RENDERER=ngl` (or gl for that matter), The layer shell works as expected. This however is not the case in KDE. It always throws an error regardless of the renderer used. I've tested using the `WAYLAND_DEBUG=1` flag and the errors are different based on the DE and the renderer used. I'll paste the last few lines where the crash happens and mention the DE on which it happens.

Hyprland - Vulkan:

```
(gtk4-layer-demo:8553): Gtk-WARNING **: 09:46:00.044: Don't know how to handle this item
[2618117.658] {Default Queue}  -> zwlr_layer_shell_v1#40.get_layer_surface(new id zwlr_layer_surface_v1#78, wl_surface#39, nil, 2, "demo")
[2618117.667] {Default Queue}  -> zwlr_layer_surface_v1#78.set_keyboard_interactivity(0)
[2618117.670] {Default Queue}  -> zwlr_layer_surface_v1#78.set_exclusive_zone(0)
[2618117.673] {Default Queue}  -> zwlr_layer_surface_v1#78.set_anchor(0)
[2618117.676] {Default Queue}  -> zwlr_layer_surface_v1#78.set_margin(0, 0, 0, 0)
[2618126.788] {Default Queue}  -> zwlr_layer_surface_v1#78.set_size(248, 266)
[2618126.802]  -> wl_surface#39.commit()
[2618127.652] {Default Queue}  -> xdg_activation_v1#15.get_activation_token(new id xdg_activation_token_v1#79)
[2618127.660]  -> xdg_activation_token_v1#79.set_serial(0, wl_seat#30)
[2618127.664]  -> xdg_activation_token_v1#79.commit()
[2618127.758] {Display Queue} wl_display#1.error(wp_linux_drm_syncobj_manager_v1#55, 0, "Surface already has a syncobj attached")
Gdk-Message: 09:46:00.054: Error 71 (Protocol error) dispatching to Wayland display.`
```

Hyprland - GL/NGL works

KDE - Vulkan:

```
(gtk4-layer-demo:9557): Gtk-WARNING **: 09:53:38.083: Don't know how to handle this item
[3076157.533] {Default Queue}  -> zwlr_layer_shell_v1#45.get_layer_surface(new id zwlr_layer_surface_v1#86, wl_surface#48, nil, 2, "demo")
[3076157.571] {Default Queue}  -> zwlr_layer_surface_v1#86.set_keyboard_interactivity(0)
[3076157.585] {Default Queue}  -> zwlr_layer_surface_v1#86.set_exclusive_zone(0)
[3076157.598] {Default Queue}  -> zwlr_layer_surface_v1#86.set_anchor(0)
[3076157.610] {Default Queue}  -> zwlr_layer_surface_v1#86.set_margin(0, 0, 0, 0)
[3076191.703] {Default Queue}  -> zwlr_layer_surface_v1#86.set_size(248, 266)
[3076191.727] {Default Queue}  -> xdg_wm_dialog_v1#23.get_xdg_dialog(new id xdg_dialog_v1#87, xdg_toplevel#6942069)
[3076191.734] {Default Queue}  -> xdg_dialog_v1#87.unset_modal()
[3076191.738]  -> wl_surface#48.commit()
[3076192.912] {Default Queue}  -> xdg_activation_v1#20.get_activation_token(new id xdg_activation_token_v1#88)
[3076192.925]  -> xdg_activation_token_v1#88.set_serial(0, wl_seat#30)
[3076192.932]  -> xdg_activation_token_v1#88.commit()
[3076193.061] {Display Queue} wl_display#1.error(wp_linux_drm_syncobj_manager_v1#64, 0, "surface already exists")
Gdk-Message: 09:53:38.119: Error 71 (Protocol error) dispatching to Wayland display.
```

KDE-NGL/GL:

```
(gtk4-layer-demo:9624): Gtk-WARNING **: 09:55:01.982: Don't know how to handle this item
[3160055.432] {Default Queue}  -> zwlr_layer_shell_v1#45.get_layer_surface(new id zwlr_layer_surface_v1#52, wl_surface#48, nil, 2, "demo")
[3160055.452] {Default Queue}  -> zwlr_layer_surface_v1#52.set_keyboard_interactivity(0)
[3160055.458] {Default Queue}  -> zwlr_layer_surface_v1#52.set_exclusive_zone(0)
[3160055.464] {Default Queue}  -> zwlr_layer_surface_v1#52.set_anchor(0)
[3160055.470] {Default Queue}  -> zwlr_layer_surface_v1#52.set_margin(0, 0, 0, 0)
[3160071.034] {Default Queue}  -> zwlr_layer_surface_v1#52.set_size(248, 266)
[3160071.056] {Default Queue}  -> xdg_wm_dialog_v1#23.get_xdg_dialog(new id xdg_dialog_v1#57, xdg_toplevel#6942069)
[3160071.066] {Default Queue}  -> xdg_dialog_v1#57.unset_modal()
[3160071.072]  -> wl_surface#48.commit()
[3160072.123] {Default Queue}  -> xdg_activation_v1#20.get_activation_token(new id xdg_activation_token_v1#58)
[3160072.134]  -> xdg_activation_token_v1#58.set_serial(0, wl_seat#30)
[3160072.142]  -> xdg_activation_token_v1#58.commit()
[3160072.337] {Display Queue} wl_display#1.delete_id(56)
[3160072.348] {Display Queue} wl_display#1.error(wl_display#1, 1, "invalid arguments for xdg_wm_dialog_v1#23.get_xdg_dialog")
Gdk-Message: 09:55:01.999: Error 22 (Invalid argument) dispatching to Wayland display.
```

It's been a notorious bug for me. I'll be happy to attach further logs (in files if they're too large ofc) if you want them.

**Comment by wmww** [comment]
I was unable to repro on Hyperland and I never saw the Vulkan bug on KDE. This could potentially be a different issue.

I was able to repro the GL bug on KDE, and I believe I fixed that in #53.

Please try out with latest main, and let me know which if any of the issues are still present. If the Vulkan issues persist, I may be able to dust off a machine with an NVIDIA GPU to test on. It would be helpful to know exactly what command/settings you're using to make it run with Vulkan.

**Comment by Roonil** [comment]
I tested this on updated KDE/Hyprland/GTK4.16, and no issues encountered this time. OpenGL/NGL/Vulkan renderers all seem to work as expected (I was just setting the `GSK_RENDERER=vulkan` flag to run GTK apps using the new Vulkan renderer). Thanks a lot!
